### PR TITLE
window title and icon support added. game can be exited now

### PR DESCRIPTION
**Pull request by Memorix101**
I added window title and icon
Exiting works now
All changes works on Windows and Unix.

### CODE REVIEW

**Comment by Menci** [comment]
Should the git settings change be merged into the repo?

**Comment by Memorix101** [comment]
it is for my fork. i edited the submodule. so when you click it in my fork you get redirected to my change.

**Comment by Hatsuney** [comment]
why would this repo refer to your fork

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,6 @@
...
-	url = https://github.com/ppy/osu-framework
-[submodule "osu-resources"]
-	path = osu-resources
-	url = https://github.com/ppy/osu-resources
+[submodule "osu-framework"]
+	path = osu-framework
+url=https://github.com/Memorix101/osu-framework
+[submodule "osu-resources"]
+	path = osu-resources
+	url = https://github.com/ppy/osu-resources
```
Comment (smoogipooo) [inline .gitmodules]: .gitmodules should not be commit, and we should probably/definitely add a .gitignore entry for this.

File: osu.Game/GameModes/Menu/ButtonSystem.cs [hunk 2]
```diff
@@ -264,6 +264,9 @@ protected override void Update()
         {
             //if (OsuGame.IdleTime > 6000 && State != MenuState.Exit)
             //    State = MenuState.Initial;
+            
+            if (state == MenuState.Exit && osuLogo.Alpha <= 0)                        
+                System.Environment.Exit(1);
```
Comment (smoogipooo) [inline osu.Game/GameModes/Menu/ButtonSystem.cs]: Game.Exit() ?

Comment (Memorix101) [inline osu.Game/GameModes/Menu/ButtonSystem.cs]: There is no Game.Exit() :P

Comment (smoogipooo) [inline osu.Game/GameModes/Menu/ButtonSystem.cs]: You will need to add to add it to Game inside the framework, but it would just call `host.Exit()`.

Comment (Memorix101) [inline osu.Game/GameModes/Menu/ButtonSystem.cs]: I replaced it

File: .gitmodules (hunk 1 above)
Comment (peppy) [inline .gitmodules]: pretty sure it should be in the repo? it's required to resolve the submodules to a repository.

just don't commit it.

File: .gitignore [hunk 3]
```diff
@@ -1,6 +1,8 @@
 ## Ignore Visual Studio temporary files, build results, and
 ## files generated by popular Visual Studio add-ons.
 
+.gitmodules
```
Comment (peppy) [inline .gitignore]: there is no point in doing this. .gitignore does *nothing* if the file is already committed.

**Pull request closed without merging**